pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -20,14 +20,20 @@ import (
 	"net/http"
 
 	genericapifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/server"
 	genericfilters "k8s.io/apiserver/pkg/server/filters"
+	"k8s.io/apiserver/pkg/util/feature"
 )
 
 // BuildInsecureHandlerChain sets up the server to listen to http. Should be removed.
 func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.Handler {
 	handler := apiHandler
-	handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
+	if feature.DefaultFeatureGate.Enabled(features.APIPriorityAndFairness) {
+		handler = genericfilters.WithPriorityAndFairness(handler, c.LongRunningFunc, c.FlowControl)
+	} else {
+		handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
```

Comment:
This existing code is very problematic if people actually use the insecure port (they shouldn't) since it doubles the number of concurrent requests. :/

Reply:
The max-in-flight filter doubles, the priority-and-fairness filter does not: line 33 passes the one shared value of `c.FlowControl`.

Reply:
the insecure handler seems boundless before.. we're actually tightening it instead doubling, isnt it?

Reply:
I think it was bounded before, but the bound was in addition to the bound for the secure chain; now both chains share the same bound/accounting.